src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1758,6 +1758,28 @@ private void BindArgumentsAndNames(BracketedArgumentListSyntax argumentListOpt,
             }
         }
 
+        private bool RefMustBeObeyed(bool isDelegateCreation, ArgumentSyntax argumentSyntax)
+        {
+            if (Compilation.FeatureStrictEnabled || !isDelegateCreation)
+            {
+                return true;
+            }
+
+            switch (argumentSyntax.Expression.Kind())
+            {
+                case SyntaxKind.ParenthesizedLambdaExpression:
+                case SyntaxKind.SimpleLambdaExpression:
+                case SyntaxKind.AnonymousMethodExpression:
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ParenthesizedExpression:
```

Comment:
Legacy compiler doesn't have a strict mode, I believe. Are you saying it always rejects any parenthesized expressions? If that is the case, please add the comment here as well because it is not common to not dig through parenthesized expressions, feels like an oversight.